addons/note/models/res_users.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+
+
+class ResUsers(models.Model):
+    _inherit = 'res.users'
+
+    @api.model
+    def create(self, data):
+        user = super(ResUsers, self).create(data)
+        NoteStage = self.env['note.stage']
+        if self.has_group('base.group_user'):
+            for n in range(5):
+                stage_id = self.sudo().env.ref('note.' + 'note_stage_%02d' % (n,), raise_if_not_found=False).id
```

Comment:
instead of this : 'note.' + 'note_stage_%02d' % (n,) ==> do this way 
'note.note_stage_%02d' % (n,)